mods/ra/rules/infantry.yaml

```diff
@@ -30,17 +31,42 @@ DOG:
 		Weapon: DogJaw
 	AttackLeap:
 		Voice: Attack
+		LeapCondition: leap
+		ApproachCondition: rush
+		RequiresCondition: !leap && !rip
 	AttackMove:
 		Voice: Move
+	GrantConditionOnAttack:
+		Condition: rip
+		RevokeDelay: 30
 	AutoTarget:
 		InitialStance: AttackAnything
 	AutoTargetPriority@DEFAULT:
 		ValidTargets: Infantry
 	Targetable:
 		TargetTypes: Ground, Infantry
 	WithInfantryBody:
-		DefaultAttackSequence: shoot
+		MoveSequence: walk
 		StandSequences: stand
+		AttackSequence: eat
+		RequiresCondition: !leap && !rush && !rip
+	WithInfantryBody@RUSH:
+		MoveSequence: run
+		AttackSequence: run
+		RequiresCondition: rush
+	WithInfantryBody@LEAP:
```

Comment:
Wouldn't it make more sense to use one of the non-Infantry sprite bodies for these?